src/mono/wasi/sample/console/Makefile

```diff
@@ -15,6 +15,9 @@ console.wasm: main.c
 	$(WASI_SDK_CLANG) main.c -o console.wasm $(COMPILE_FLAGS) $(LINK_FLAGS)
 
 $(BIN_DIR)/WasiConsoleApp.dll: WasiConsoleApp/*.csproj WasiConsoleApp/*.cs
-	cd WasiConsoleApp && $(DOTNET_ROOT)/dotnet build -c $(CONFIG)
+	cd WasiConsoleApp && $(DOTNET_ROOT)/dotnet build -c Debug
 	touch $(BIN_DIR)/*.dll
 	cp -R $(BROWSER_WASM_RUNTIME_PATH) $(BIN_DIR)/runtime
+
+debug: all
+	wasmtime --tcplisten localhost:64000 --dir=. console.wasm --env DEBUGGER_FD=4
```

Comment:
What's the FD=4 here?

Reply:
I'm a bit unsure because with WASI conventions the preopened TCP sockets start at FD=3, and it looks like there's only one preopen here (port 64000) so I would have expected `DEBUGGER_FD=3`. Hopefully @thaystg can clarify!

Reply:
Because we are passing --dir=. so the directory is the FD=3 and the socket is FD=4.

Reply:
Oh interesting. I thought wasmtime was hardcoded to put the TCP sockets first. If it works this way round, that makes sense too.

Reply:
Right now it doesn't even work with wasmtime, because there is an error if you try to use the directory and the socket parameter at same time. But if I build wasmtime locally using the PR that is open on their repo with the fix, it works like this.

Reply:
Ok, this sounds fine for now. Can you add a comment explaining this though?